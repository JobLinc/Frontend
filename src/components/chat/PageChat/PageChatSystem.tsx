import { useCallback, useEffect, useState } from "react";
import ChatCardsList from "../ChatCardsList";
import PageMessageWindow from "./PageChatWindow";
import useChatId from "@context/ChatIdProvider";
import connectToChat, { disconnectChatSocket } from "@services/api/ChatSocket";
import { EllipsisVertical } from "lucide-react";
import SearchBar from "@chatComponent/UI/SearchBar";
import NetWorksChatList from "@chatComponent/NetWorksChatList";
import useNetworkUserId from "@context/NetworkUserIdProvider";
import ConnectionsSidebar from "@chatComponent/PageChat/ConnectionsSidebar";
import { Users2 } from "lucide-react";

function PageChatSystem() {
  const { setChatId } = useChatId();
  const { setUsersId } = useNetworkUserId();
  console.log("Page System");
  const [isFocused, setIsFocused] = useState<boolean>(false);
  const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(false);
  const onFocusedToggler = () => {
    setTimeout(() => {
      setIsFocused(!isFocused);
    }, 100);
  };
  const handleSearchChange = useCallback((value: string) => {
    console.log(value);
  }, []);
  const handleConversationClick = useCallback((chatId: string) => {
    setUsersId([]);
    setChatId(chatId);
  }, []);
  const handleNetWorkUserClick = useCallback((userId: string) => {
    setUsersId([userId]);
    setChatId("");
  }, []);

  useEffect(() => {
    connectToChat();
    return () => disconnectChatSocket();
  }, []);
  return (
    <>
      <header className="flex items-center justify-between bg-gray-100 p-3 rounded-lg shadow-md">
        <h2 className=" hidden md:block font-semibold text-lg text-gray-800">
          Messaging
        </h2>

        <div className="flex items-center gap-3">
          <SearchBar
            FocusToggler={onFocusedToggler}
            onChange={handleSearchChange}
            className="md:w-xl "
          />
          <button
            onClick={() => setIsSidebarOpen(true)}
            className="p-2 rounded-full hover:bg-gray-200 transition"
          >
            <Users2 />
          </button>
          {isSidebarOpen && (
            <ConnectionsSidebar
              isOpen={isSidebarOpen}
              onClose={() => setIsSidebarOpen(false)}
            />
          )}
          <button className="p-2 rounded-full hover:bg-gray-200">
            <EllipsisVertical className="w-6 h-6 text-gray-600" />
          </button>
        </div>
      </header>
      <div className="flex h-screen w-full">
        <div className="w-1/3 border-r bg-charcoalWhite border-gray-300 p-4 overflow-y-auto">
          {isFocused ? (
            <NetWorksChatList
              className=""
              onCardClick={handleNetWorkUserClick}
            />
          ) : (
            <ChatCardsList className="" onCardClick={handleConversationClick} />
          )}
        </div>
        <PageMessageWindow className="flex-grow flex flex-col justify-end" />
      </div>
    </>
  );
}

export default PageChatSystem;
