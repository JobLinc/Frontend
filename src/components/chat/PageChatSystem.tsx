import React, { useState } from "react";
import ChatCardsList from "./ChatCardsList";
import PageMessageWindow from "./PageChatWindow";

export default function PageChatSystem() {
  const [selectedChatId, setSelectedChatId] = useState<string | null>(null);
  console.log("--------------PageChatSystem rendered--------------");
  const handleConversationClick = (chatId: string) => {
    setSelectedChatId(chatId);
  };
  return (
    <div className="flex h-screen w-full">
      <div className="w-1/3 border-r bg-charcoalWhite border-gray-300 p-4 overflow-y-auto">
        <div className="flex items-center gap-4 mb-4">
          <h2 className="font-semibold text-lg">Messaging</h2>
        </div>
        <ChatCardsList onCardClick={handleConversationClick} />
      </div>
      <PageMessageWindow
        className="flex-grow flex flex-col justify-end"
        chatId={selectedChatId}
      />
    </div>
  );
}
