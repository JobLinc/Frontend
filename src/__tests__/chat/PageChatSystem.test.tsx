import { render } from "@testing-library/react";
import { describe, it, expect, vi, afterEach } from "vitest";
import * as React from "react";
import PageChatSystem from "@chatComponent/PageChat/PageChatSystem";

// Mock useState
const DUMMY_FN = () => {};
const states = [
  [false, DUMMY_FN], // isFocused
  [false, DUMMY_FN], // isSidebarOpen
  [true, DUMMY_FN], // isChatSidebarOpen
  ["", DUMMY_FN], // searchTerm
  ["", DUMMY_FN], // opnedChatName
  [true, DUMMY_FN], // isConnected
  [false, DUMMY_FN], // isMobileView
  [false, DUMMY_FN], // menuOpen
  [false, DUMMY_FN], // showBlockConfirmation
];

describe("PageChatSystem", () => {
  it("renders ChatCardsList with mocked useState", () => {
    render(<PageChatSystem />);
  });
});
