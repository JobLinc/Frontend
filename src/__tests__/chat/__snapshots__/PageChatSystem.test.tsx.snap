// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageChatSystem  > Should RenderChatSystem:  1`] = `
<div>
  <div
    class="flex h-screen w-full"
  >
    <div
      class="w-1/3 border-r bg-charcoalWhite border-gray-300 p-4 overflow-y-auto"
    >
      <div
        class="flex items-center gap-4 mb-4"
      >
        <h2
          class="font-semibold text-lg"
        >
          Messaging
        </h2>
      </div>
      <div
        class="mt-4"
      >
        <div
          class="text-gray-500 text-center mt-4"
        >
          No chats found
        </div>
      </div>
    </div>
    <div
      class="flex-grow flex flex-col justify-end w-2/3 flex flex-col"
    >
      <div
        class="flex-grow flex items-center justify-center"
      >
        <h2
          class="text-gray-500"
        >
          Choose a chat to start messaging
        </h2>
      </div>
    </div>
  </div>
</div>
`;
